grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -2708,7 +2707,9 @@ class InventoryService implements ApplicationContextAware {
                 }
             }
             transaction {
-                lt("transactionDate", date)
+                // We are able to select dates with the precision set to minutes,
+                // so we should have at most one transaction within the same minute
+                lt("transactionDate",  new Date(date.time + 60_000))
```

Comment:
I don't think this is what we want to do universally. In cycle count we use current timestamp for transaction date, which is precise to the second. So this logic is overly restrictive for that case.

But also, if a date picker only gives minute level precision, any transaction date that uses that picker within the same minute should get a date that is the start of the minute. I don't see why we need to consider any other times within that minute unless we're worried about other flows changing the same products at the same time that have precision to the second.

Is the issue not just that we're missing a baseline if there is no stock previously so this duplicate check doesn't fail? Can we not just change the logic to create the baseline for the default lot and bin with quantity 0?